### PR TITLE
[SPARK-41020][SQL] Rename the error class `_LEGACY_ERROR_TEMP_1019` to `STAR_GROUP_BY_POS`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to rename the legacy error class `_LEGACY_ERROR_TEMP_1019` to `STAR_GROUP_BY_POS`, and improve the error message format slightly.

### Why are the changes needed?
Proper names of error classes should improve user experience with Spark SQL.

### Does this PR introduce _any_ user-facing change?
Yes.

### How was this patch tested?
By running the affected test suites:
```
$ build/sbt "core/testOnly *SparkThrowableSuite"
$ PYSPARK_PYTHON=python3 build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite -- -z group-by-ordinal.sql"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @srielau @itholic @LuciferYang @panbingkun Please, review this PR.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -912,6 +912,11 @@
     ],
     "sqlState" : "22023"
   },
+  "STAR_GROUP_BY_POS" : {
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:915]: LGTM ... but why do we have this restriction?

**Review by srielau — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:915]: It is better to ask the author of the restriction, cc @gatorsmile 
https://github.com/apache/spark/pull/11846/files#diff-ed19f376a63eba52eea59ca71f3355d4495fad4fad4db9a3324aade0d4986a47R389-R392

**Review by itholic — APPROVED** [review]
LGTM

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan @LuciferYang @itholic @srielau.

**Pull request closed without merging**